raspberrypi_central/webapp/app/alarm/models.py

```diff
@@ -1,28 +1,141 @@
+                task='alarm.set_alarm_on',
+                clocked=clocked_turn_on,
+                one_off=True
+            )
+
+            # @TODO: turn off with another task:
+            # 1) off the alarm via mqtt 2) THEN delete this!
+            clocked_turn_off = ClockedSchedule.objects.create(clocked_time=self.datetime_end)
+            self.turn_off_task = PeriodicTask.objects.create(
+                name=f'Turn off alarm for holidays {uid}',
+                task='alarm.set_alarm_off',
+                clocked=clocked_turn_off,
+                one_off=True
+            )
+        else:
+            on_clock = self.turn_on_task.clocked
+            off_clock = self.turn_off_task.clocked
+
+            on_clock.clocked_time = self.datetime_start
+            off_clock.clocked_time = self.datetime_end
+
+            on_clock.save()
+            off_clock.save()
+
+        super().save(*args, **kwargs)
+
+    def delete(self):
+        schedules = AlarmSchedule.objects.filter(is_disabled_by_system=True)
+
+        for schedule in schedules:
+            schedule.enable()
+
+        super(AlarmScheduleDateRange, self).delete()
+
+def create_celery_schedules(uid: str, timezone: str, cron_days: str, hour_start, minute_start, hour_end, minute_end):
+
+    schedule_turn_on_alarm = CrontabSchedule.objects.create(
+        minute=minute_start,
+        hour=hour_start,
+        day_of_week=cron_days,
+        timezone=timezone
+    )
+
+    turn_on_task = PeriodicTask.objects.create(
+        name=f'Turn on alarm {uid}',
+        task='alarm.set_alarm_on',
+        crontab=schedule_turn_on_alarm
+    )
+
+    schedule_turn_off_alarm = CrontabSchedule.objects.create(
+        minute=minute_end,
+        hour=hour_end,
+        day_of_week=cron_days,
+        timezone=timezone
+    )
+
+    turn_off_task = PeriodicTask.objects.create(
+        name=f'Turn off alarm {uid}',
+        task='alarm.set_alarm_off',
+        crontab=schedule_turn_off_alarm
+    )
+
+    return {'turn_off_task': turn_off_task, 'turn_on_task': turn_on_task}
+
 class AlarmSchedule(models.Model):
     hour_start = models.IntegerField()
     minute_start = models.IntegerField()
 
     hour_end = models.IntegerField()
     minute_end = models.IntegerField()
 
-    monday    = models.BooleanField()
-    tuesday   = models.BooleanField()
-    wednesday = models.BooleanField()
-    thursday  = models.BooleanField()
-    friday    = models.BooleanField()
-    saturday  = models.BooleanField()
-    sunday    = models.BooleanField()
+    monday    = models.BooleanField(default=False)
+    tuesday   = models.BooleanField(default=False)
+    wednesday = models.BooleanField(default=False)
+    thursday  = models.BooleanField(default=False)
+    friday    = models.BooleanField(default=False)
+    saturday  = models.BooleanField(default=False)
+    sunday    = models.BooleanField(default=False)
+
+    is_disabled_by_system = models.BooleanField(default=False)
```

Comment:
Not needed because during a date range schedule, we disable all others schedules.